File: fizz_buzz.rb
def is_fizz(num:)
  return num % 3 == 0 ? 0b1 : 0b0
end

def is_buzz(num:)
  return num % 5 == 0 ? 0b1 : 0b0
end

def fizz_buzz(max:)
  (1..max).each do |num|
    case (is_fizz(num: num) + is_buzz(num: num) * 2)
    in 0b11
      puts "#{num} is FizzBuzz"
    in 0b10
      puts "#{num} is Buzz"
    in 0b01
      puts "#{num} is Fizz"
    else
      puts num
    end
  end
end

fizz_buzz(max: 50)

File: fizz_buzz.zig
const std = @import("std");

fn fizzBussIf(max: u8) void {
    for (1..max + 1) |num| {
        if (num % 15 == 0) {
            std.debug.print("{} is FizzBuzz\n", .{num});
        } else if (num % 5 == 0) {
            std.debug.print("{} is Buzz\n", .{num});
        } else if (num % 3 == 0) {
            std.debug.print("{} is Fizz\n", .{num});
        } else {
            std.debug.print("{}\n", .{num});
        }
    }
}

fn fizzBussSwitch(max: u8) void {
    for (1..(max + 1)) |num| {
        const is_fizz: u2 = @intFromBool(num % 3 == 0);
        const is_buzz: u2 = @intFromBool(num % 5 == 0);
        switch (is_fizz + is_buzz * 2) {
            0b01 => {
                std.debug.print("{} is Fizz\n", .{num});
            },
            0b10 => {
                std.debug.print("{} is Buzz\n", .{num});
            },
            0b11 => {
                std.debug.print("{} is FizzBuzz\n", .{num});
            },
            else => {
                std.debug.print("{}\n", .{num});
            },
        }
    }
}

pub fn main() void {
    fizzBussIf(50);
    fizzBussSwitch(50);
}
